### PR TITLE
Color cycles properties return array

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
As discussed in #1104 , this PR changes the color cycle properties to only be set by lists a numpy arrays of colors and the property returns the same way. This change is to enable serialized layers to be be written out in a human-readable format.

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# References

# How has this been tested?
- [x] all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/layers/points/points.py [hunk 1]
```diff
@@ -696,23 +696,15 @@ def edge_color(self, edge_color):
 
     @property
     def edge_color_cycle(self):
-        """Union[list, np.ndarray, cycle] :  Color cycle for edge_color.
-        Can be a list of colors or a cycle of colors
+        """Union[list, np.ndarray] :  Color cycle for edge_color.
```
Comment (tlambert03) [inline napari/layers/points/points.py]: maybe while we're at it we can turn these into proper return annotations?

```python
    def edge_color_cycle(self) -> Union[list, np.ndarray]:
        """Color cycle for edge_color.

        Can be a list of colors defined by name, RGB or RGBA.
        """
```

(same with other similar functions)

File: napari/layers/utils/color_transformations.py [hunk 2]
```diff
@@ -74,20 +74,18 @@ def transform_color_cycle(
     -------
     transformed : cycle
         cycle of Nx4 numpy arrays with a dtype of np.float32
+    transformed_color_cycle: np.ndarray
```
Comment (tlambert03) [inline napari/layers/utils/color_transformations.py]: does the return annotation need to be updated here?

**Review by tlambert03 — APPROVED** [review]
thanks @kevinyamauchi!

File: napari/layers/utils/color_transformations.py (hunk 2 above)
Comment (kevinyamauchi) [reply napari/layers/utils/color_transformations.py]: Good point. I also think the variable names were confusing so I updated them.

File: napari/layers/points/points.py (hunk 1 above)
Comment (kevinyamauchi) [reply napari/layers/points/points.py]: I think it makes sense (it's been on the to do list). I can work on it later today!

Comment (kevinyamauchi) [reply napari/layers/points/points.py]: Done!

**Comment by kevinyamauchi** [comment]
Thanks for the quick review, @tlambert03 ! @sofroniewn , would you mind taking a look to make sure this matches what you need for the #1104 ?

**Review by sofroniewn — APPROVED** [review]
LGTM! Thanks for the rapid turn around on this one @kevinyamauchi! Will give @jni the option to review and then merge!

**Review by jni — APPROVED** [review]

**Pull request merged**